docs/docs/configuration/pages.md

```diff
@@ -95,7 +97,17 @@ export default function SignIn({ providers }) {
 }
 
 export async function getServerSideProps(context) {
-  const providers = await getProviders()
+  const session = await getServerSession(context.req, context.res, authOptions);
+  
+  // If sign in is successful, redirect to homepage
+  if (session) {
```

Comment:
Actually, we should probably also check that the destination is not the same as the sign-in page, otherwise, you might end up in an infinite loop.